auth/credentials/impersonate/impersonate.go

```diff
@@ -87,30 +90,32 @@ func NewCredentials(opts *CredentialsOptions) (*auth.Credentials, error) {
 		client = opts.Client
 	}
 
+	universeDomainProvider := resolveUniverseDomainProvider(opts, creds)
 	// If a subject is specified a domain-wide delegation auth-flow is initiated
 	// to impersonate as the provided subject (user).
 	if opts.Subject != "" {
-		if !opts.isUniverseDomainGDU() {
+		gdu, err := isUniverseDomainGDU(universeDomainProvider)
```

Comment:
Can this check be moved to the impl for the token provider? Having this code here puts this check at construction time vs usage-time.

Reply:
Fixed, thanks!